modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v9_0_1/ObjectFolderUpgradeProcess.java

```diff
@@ -54,6 +54,18 @@ protected void doUpgrade() throws Exception {
 						Role guestRole = _roleLocalService.getRole(
 							company.getCompanyId(), RoleConstants.GUEST);
 
+						if (_resourcePermissionLocalService.
+								hasResourcePermission(
+									company.getCompanyId(),
+									ObjectFolder.class.getName(),
+									ResourceConstants.SCOPE_INDIVIDUAL,
+									String.valueOf(
+										resultSet.getLong("objectFolderId")),
+									guestRole.getRoleId(), ActionKeys.VIEW)) {
+
+							continue;
+						}
+
```

Comment:
@pedro-oliveira446 Is this really necessary? I believe setResourcePermissions can be executed more than once.

cc @MarinhoFeliphe

Reply:
@gabrielwas I discussed this with @MarinhoFeliphe  and we agree with you. Thanks.